metrics/setup.go

```diff
@@ -0,0 +1,142 @@
+package metrics
+
+import (
+	"github.com/packethost/pkg/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	DHCPTotal *prometheus.CounterVec
+
+	CacherDuration           prometheus.ObserverVec
+	CacherCacheHits          *prometheus.CounterVec
+	CacherTotal              *prometheus.CounterVec
+	CacherRequestsInProgress *prometheus.GaugeVec
+
+	DiscoverDuration    prometheus.ObserverVec
+	HardwareDiscovers   *prometheus.CounterVec
+	DiscoversInProgress *prometheus.GaugeVec
+
+	JobDuration    prometheus.ObserverVec
+	JobsTotal      *prometheus.CounterVec
+	JobsInProgress *prometheus.GaugeVec
+)
+
+func Init(_ log.Logger) {
+	DHCPTotal = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "dhcp_total",
+		Help: "Number of DHCP Requests handled.",
+	}, []string{"op", "type", "giaddr"})
+
+	labelValues := []prometheus.Labels{
+		{"op": "recv", "type": "DHCPACK", "giaddr": "0.0.0.0"},
+		{"op": "recv", "type": "DHCPDECLINE", "giaddr": "0.0.0.0"},
+		{"op": "recv", "type": "DHCPDISCOVER", "giaddr": "0.0.0.0"},
+		{"op": "recv", "type": "DHCPINFORM", "giaddr": "0.0.0.0"},
+		{"op": "recv", "type": "DHCPNAK", "giaddr": "0.0.0.0"},
+		{"op": "recv", "type": "DHCPOFFER", "giaddr": "0.0.0.0"},
+		{"op": "recv", "type": "DHCPRELEASE", "giaddr": "0.0.0.0"},
+		{"op": "recv", "type": "DHCPREQUEST", "giaddr": "0.0.0.0"},
+		{"op": "send", "type": "DHCPOFFER", "giaddr": "0.0.0.0"},
```

Comment:
missing `{"op": "send", "type": "DHCPACK", "giaddr": "0.0.0.0"},`
also :thinking: how does a different giaddr not blow up?

Reply:
This list will definitely grow. But it should typically only grow to as many relays as you have pointed to it. I thought this would be a good way to potentially see hotspots in requests.